File: ClientJava/network/src/main/java/network/udp/ProtocolEventsUDP.java
package network.udp;

import network.message.obj.Choice;
import network.message.obj.DataServerGame;
import network.protocol.ProtocolEvents;

public class ProtocolEventsUDP<T> extends ProtocolEvents<T> {

    public static ProtocolEventsUDP<Choice> SUBSCRIPTION = new ProtocolEventsUDP<Choice>("SUBSCRIPTION");
    public static ProtocolEventsUDP<DataServerGame> GETLISTSERVERGAME = new ProtocolEventsUDP<>("GETLISTSERVERGAME");
    public static ProtocolEventsUDP<String> NEWGAME = new ProtocolEventsUDP<String>("NEWGAME");
    public static ProtocolEventsUDP<String> REMOVEGAME = new ProtocolEventsUDP<String>("REMOVEGAME");
    private ProtocolEventsUDP(String name){
        super(name);
    }




}

File: ServeurWheelOfFortune/share/Network/UDP/ProtocolEventsUDP.cs
using share;
using Share.Network.Message.obj;
using System;
using System.Collections.Generic;
using System.Text;

namespace Share.Network.Protocol
{
    public class ProtocolEventsUDP<T> : ProtocolEvents<T>
    {

        public static ProtocolEventsUDP<Choice> SUBSCRIPTION = new ProtocolEventsUDP<Choice>("SUBSCRIPTION");
        public static ProtocolEventsUDP<DataServerGame> GETLISTSERVERGAME = new ProtocolEventsUDP<DataServerGame>("GETLISTSERVERGAME");
        public static ProtocolEventsUDP<String> NEWGAME = new ProtocolEventsUDP<String>("NEWGAME");
        public static ProtocolEventsUDP<String> REMOVEGAME = new ProtocolEventsUDP<String>("REMOVEGAME");
        private ProtocolEventsUDP(String name) : base(name)
        {

        }

    }
}
